### PR TITLE
Use Packer to build an AMI with Screwdriver service

### PR DESCRIPTION
**Pull request by jer**

### CODE REVIEW

**Comment by jer** [comment]
Wercker.yml needs to point to the new app directory

File: config/packer.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "variables": {
+      "troy_base_ami": "ami-f119e591"
```
Comment (jer) [inline config/packer.json]: We should use a different base. Ubuntu 15.10

File: config/packer.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  "builders": [{
+    "type": "amazon-ebs",
+    "region": "us-west-2",
+    "source_ami": "{{user `troy_base_ami`}}",
```
Comment (jer) [inline config/packer.json]: Call this `base_ami`

File: config/setup.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+# download node
+curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -
+sudo apt-get install -y nodejs
+
+# move application to proper location
+APP_LOCATION=$HOME/screwdriver
+
+mkdir -p $APP_LOCATION
+cp -r /tmp/application/* $APP_LOCATION
```
Comment (jer) [inline config/setup.sh]: mv

File: config/setup.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+sudo apt-get -y update
+
+# download node
+curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -
+sudo apt-get install -y nodejs
+
+# move application to proper location
+APP_LOCATION=$HOME/screwdriver
```
Comment (jer) [inline config/setup.sh]: This should probably be `/usr/src/app` (Standard node app location)

Comment (jer) [inline config/setup.sh]: Needs to sudo mkdir the directory first.

Comment (jer) [inline config/setup.sh]: Change owner to ubuntu

Comment (jer) [inline config/setup.sh]: Give read permissions to `nobody`

**Comment by jer** [comment]
Make init script

**Comment by jer** [comment]
Squash all changes when done

File: config/setup.sh [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+sudo apt-get -y update
+
+# download node
+curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -
+sudo apt-get install -y nodejs
+
+# move application to proper location
+APP_LOCATION=/usr/src/app
```
Comment (nkatzman) [inline config/setup.sh]: do we need to mkdir -p this directory?

Comment (nkatzman) [inline config/setup.sh]: just pulled this down and tried locally. Looks like the directory already exists

**Comment by stjohnjohnson** [comment]
LGTM 👍 
excellent

**Comment by nkatzman** [comment]
still need init script stjohn ^^^^^^^

**Comment by FenrirUnbound** [comment]
👍 

Looks great to me

**Comment by nkatzman** [comment]
👍  worth

File: config/packer.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    "associate_public_ip_address": true
+  }],
+  "provisioners": [
+  {
+    "type": "file",
+    "source": "app",
+    "destination": "/tmp/application"
+  },
+  {
+      "type": "file",
```
Comment (nkatzman) [inline config/packer.json]: spacing is off

File: config/packer.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
+{
+  "variables": {
+      "base_ami": "ami-2439c444"
```
Comment (nkatzman) [inline config/packer.json]: spaaaaace https://www.youtube.com/watch?v=zBYC4f79iXc

**Comment by nkatzman** [comment]
👍  suh dude

**Pull request merged**